mirgecom/flux.py

```diff
@@ -35,179 +40,224 @@
+    and exterior of the face on which the central flux is to be calculated, and
+    $\mathbf{\hat{n}}$ is the *normal* vector. Numerical dissipation is added
+    by the solution penalized by a factor $\beta$.
+
+    *u_tpair* is the :class:`~grudge.trace_pair.TracePair` representing the scalar
+    quantities ${u}^-, {u}^+$. *u_tpair* may also represent a vector-quantity
+    :class:`~grudge.trace_pair.TracePair`, and in this case the central scalar flux
+    is computed on each component of the vector quantity as an independent scalar.
 
     Parameters
     ----------
-    cv_tpair: :class:`~grudge.trace_pair.TracePair`
+    u_tpair: :class:`~grudge.trace_pair.TracePair`
+        Trace pair for the face upon which flux calculation is to be performed
+    beta: Number or :class:`~meshmode.dof_array.DOFArray`
+        Optional dissipation strength term
+    normal: numpy.ndarray
+        object array of :class:`~meshmode.dof_array.DOFArray` with outward-pointing
+        normals
+
+    Returns
+    -------
+    numpy.ndarray
+        object array of :class:`~meshmode.dof_array.DOFArray` with the flux for each
+        scalar component.
+    """
+    from arraycontext import outer
+    return outer(u_tpair.avg + .5*beta*u_tpair.diff, normal)
 
-        Solution trace pair for faces for which numerical flux is to be calculated
 
-    f_tpair: :class:`~grudge.trace_pair.TracePair`
+def divergence_flux(trace_pair, normal, alpha=0, beta=0):
+    r"""Compute a numerical flux for the divergence operator.
 
-        Physical flux trace pair on faces on which numerical flux is to be calculated
+    The divergence flux, $h$, is calculated as:
 
-    normal: numpy.ndarray
+    .. math::
 
-        object array of :class:`~meshmode.dof_array.DOFArray` with outward-pointing
-        normals
+        h(\mathbf{v}^-, \mathbf{v}^+; \mathbf{n}) = \frac{1}{2}
+        \left(\mathbf{v}^{+}+\mathbf{v}^{-}+\beta(\mathbf{v}^{+}-\mathbf{v}^{-}
+        +\alpha\right) \cdot \hat{n}
 
-    lam: :class:`~meshmode.dof_array.DOFArray`
+    where $\mathbf{v}^-, \mathbf{v}^+$, are the vectors on the interior and exterior
+    of the face across which the central flux is to be calculated, and $\hat{n}$ is
+    the unit normal to the face.  \alpha and \beta are optional terms for adding
+    dissipation to the otherwise central flux.
 
-        lambda parameter for Lax-Friedrichs/Rusanov flux
+    Parameters
+    ----------
+    trace_pair: :class:`~grudge.trace_pair.TracePair`
+        Trace pair for the face upon which flux calculation is to be performed
+    normal: numpy.ndarray
+        object array of :class:`~meshmode.dof_array.DOFArray` with outward-pointing
+        normals
 
     Returns
     -------
     numpy.ndarray
-
-        object array of :class:`~meshmode.dof_array.DOFArray` with the
-        Lax-Friedrichs/Rusanov flux.
+        object array of :class:`~meshmode.dof_array.DOFArray` with the flux for each
+        scalar component.
```

Comment:
This doesn't look quite right. Maybe we should be intentionally vague here since the function is generic and the result type depends on what's inside the trace pair?

Reply:
Almost -- I would get rid of the `numpy.ndarray` result type (since that's likely not what it will be) and the object array language in the description. Maybe just say something along the lines of `object of the same type as *trace_pair.int* containing the face-normal fluxes for each component`?